### PR TITLE
MAGECLOUD-4567: De-compose All Patches from ECE-Tools

### PR DESCRIPTION
**Pull request by shiftedreality**
### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-4567

### Manual testing scenarios

1. Edit `composer.json`:
```json
    "repositories": {
        "repo": {
            "type": "composer",
            "url": "https://repo.magento.com"
        },
        "ece-tools": {
            "type": "vcs",
            "url": "git@github.com:magento/ece-tools.git"
        },
        "mcp": {
            "type": "vcs",
            "url": "git@github.com:magento/magento-cloud-patches.git"
        }
    },
```

```json
    "require": {
        "magento/magento-cloud-metapackage": ">=2.3.0 <2.3.1",
        "magento/ece-tools": "dev-MAGECLOUD-4567 as 2002.0.99",
        "magento/magento-cloud-patches": "dev-develop as 1.0.0"
    },
```
2. Perform acceptance tests

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,10 +28,10 @@
     "symfony/dependency-injection": "^3.4||^4.3",
     "symfony/process": "^2.1||^4.1",
     "symfony/serializer": "^3.3",
-    "symfony/yaml": "^3.3||^4.0"
-  },
-  "suggest": {
-    "magento/magento-cloud-docker": "^1.0"
+    "symfony/yaml": "^3.3||^4.0",
+    "magento/magento-cloud-components": "^1.0.1",
```
Comment (oshmyheliuk) [inline composer.json]: Why do you move `magento-cloud-components` to suggest? Ece-tools depends on CLI commands from that package.

File: scenario/build/generate.xml [hunk 2]
```diff
@@ -23,6 +23,7 @@
         </arguments>
     </step>
     <step name="refresh-modules" type="Magento\MagentoCloud\Step\Build\RefreshModules" priority="400"/>
+    <step name="copy-pub-static" type="Magento\MagentoCloud\Command\CopyPubStatic" priority="450"/>
```
Comment (oshmyheliuk) [inline scenario/build/generate.xml]: Now we can change here priority to 500 and add +100 to the next steps, as we didn't release yet.

File: composer.json (hunk 1 above)
Comment (shiftedreality) [reply composer.json]: It's in `require`, just weird GitHub visual

File: composer.json [hunk 3]
```diff
@@ -29,10 +28,10 @@
...
+    "magento/magento-cloud-docker": "^1.0.0",
+    "magento/magento-cloud-patches": "^1.0.0"
```
Comment (bbatsche) [inline composer.json]: Ideally we should keep these alphabetized, but that's minor

File: src/Patch/Manager.php [hunk 4]
```diff
@@ -7,171 +7,68 @@
...
-    private function applyHotFixes()
-    {
-        $hotFixesDir = $this->directoryList->getMagentoRoot() . '/' . static::HOTFIXES_DIR;
-
-        if (!$this->file->isDirectory($hotFixesDir)) {
-            $this->logger->notice('Hot-fixes directory was not found. Skipping.');
 
-            return;
+        try {
+            $this->shell->execute($command)->getOutput();
```
Comment (bbatsche) [inline src/Patch/Manager.php]: Should this be logged? If not, why `getOutput()`?

**Review by bbatsche — CHANGES_REQUESTED** [review]
Just want some clarity on `getOutput()` question, otherwise looks good.

File: scenario/build/generate.xml (hunk 2 above)
Comment (shiftedreality) [reply scenario/build/generate.xml]: Fixed

File: composer.json (hunk 3 above)
Comment (shiftedreality) [reply composer.json]: Done

File: src/Patch/Manager.php (hunk 4 above)
Comment (shiftedreality) [reply src/Patch/Manager.php]: The output is logged always in Shell, removed `getOuptput`

**Review by bbatsche — DISMISSED** [review]

**Review by oshmyheliuk — DISMISSED** [review]

**Review by bbatsche — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**